app/src/middleware/authorization.js

```diff
@@ -44,6 +48,54 @@ const _checkPermission = async ({ currentObject, currentUser, params }, permissi
   log.debug('Missing user identification', { function: '_checkPermission' });
   return result;
 };
+/**
+ * @function checkS3BasicAccess
+ * Checks and authorized access to perform operation for s3 basic authentication request
+ * @param {object} req Express request object
+ * @param {object} _res Express response object
+ * @param {function} next The next callback function
+ * @returns {function} Express middleware function
+ * @throws The error encountered upon failure
+ * @returns
+ */
+const checkS3BasicAccess = async (req, _res, next) => {
+  const authType = req.currentUser ? req.currentUser.authType : undefined;
+  const bucketSettings = req.currentUser?.bucketSettings ? req.currentUser.bucketSettings : undefined;
+  let bucketIds = mixedQueryToArray(req.query.bucketId) || mixedQueryToArray(req.params.bucketId) || req.body.bucketId;
+  const objIds = mixedQueryToArray(req.query.objectId) || mixedQueryToArray(req.params.objectId) || req.body.objectId;
+
+  if (!bucketIds?.length && objIds?.length) {
+    const objectIds = await objectService.searchObjects({ id: objIds });
+    bucketIds = objectIds.data.map(i => i.bucketId);
+  }
+  if (
+    authType === AuthType.BASIC &&
+    bucketSettings) {
+    try {
+      const bucketData = {
+        bucketId: bucketIds,
+        bucket: bucketSettings.bucket,
+        endpoint: bucketSettings.endpoint,
+        accessKeyId: bucketSettings.accessKeyId,
+      };
+      const buckets = await bucketService.checkBucketBasicAccess(bucketData);
+
+      if (buckets.length === 0) {
+        return next(new Problem(403, {
+          detail: 'User lacks permission to complete this action',
```

Comment:
maybe replace with 
`          detail: 'The provided S3 credentials do not authorize this operation',
`
user doesnt really make sense in s3 access mode